keras/engine/input_layer.py

```diff
@@ -107,9 +115,32 @@ def get_config(self):
         config = {'batch_input_shape': self.batch_input_shape,
                   'dtype': self.dtype,
                   'sparse': self.sparse,
-                  'name': self.name}
+                  'name': self.name,
+                  'is_placeholder': self.is_placeholder}
+
+        # If the input layer is not a placeholder,
+        #  store the value of the layer assigned by the users
+        if not self.is_placeholder:
+            # TODO (wardlt): Would it be better to store using add_weights?
+            config['user_provided_weights'] = \
+                K.eval(self._predefined_input_tensor).tolist()
         return config
 
+    @classmethod
+    def from_config(cls, config):
+        # Remove `is_placeholder` and from the configuration
+        #  as they are not part of the constructor
+        config = dict(config)
+        is_placeholder = config.pop('is_placeholder', True)
```

Comment:
E.g. this change would have to be duplicated in every library that is capable of loading Keras models. This is difficult.